web/client/reducers/draw.js

```diff
@@ -49,7 +50,21 @@ function draw(state = initialState, action) {
             currentStyle: action.currentStyle
         });
     case GEOMETRY_CHANGED:
-        return assign({}, state, {tempFeatures: action.features});
+        let newData = action.features;
```

Comment:
since you mutate the newData later maybe is better to clone it before
```suggestion
        let newData = assign({},action.features);
```